packages/http-server/test/integration/http-server.integration.ts

```diff
@@ -184,18 +195,51 @@ describe('HttpServer (integration)', () => {
 
   it('converts host from [::] to [::1] in url', async () => {
     // Safari on MacOS does not support http://[::]:3000/
-    server = new HttpServer(dummyRequestHandler, {host: '::'});
+    server = givenHttpServer(dummyRequestHandler, {host: '::'});
     await server.start();
     expect(server.url).to.equal(`http://[::1]:${server.port}`);
   });
 
   it('converts host from 0.0.0.0 to 127.0.0.1 in url', async () => {
     // Windows does not support http://0.0.0.0:3000/
-    server = new HttpServer(dummyRequestHandler, {host: '0.0.0.0'});
+    server = givenHttpServer(dummyRequestHandler, {host: '0.0.0.0'});
     await server.start();
     expect(server.url).to.equal(`http://127.0.0.1:${server.port}`);
   });
 
+  it('supports HTTP/2 protocol with key and certificate files', async () => {
+    // For some reason, spdy fails with Node 11
+    if (+process.versions.node.split('.')[0] > 10) return;
+    const serverOptions: Http2Options = Object.assign(
+      {
+        protocol: 'http2' as 'http2',
```

Comment:
```ts
'http2' as 'http2'
```

This looks weird, are you sure it's needed?

Reply:
If we don't cast `'http2'` to `'http2'`, TS complains as it infers the type of `http2` as `string`, which is not compatible with `'http' | 'https' | 'http2'`.

Reply:
Weird.

Can you add a comment to explain to readers please? (Similarly for 'h2' below.)